frontend/src/modules/Intersection/settings/settings.tsx

```diff
@@ -29,35 +31,29 @@ export function Settings(props: ModuleSettingsProps<Interfaces>): JSX.Element {
     const colorSet = useColorSet(props.workbenchSettings);
 
     const [dataProviderManager, setDataProviderManager] = useAtom(dataProviderManagerAtom);
+    const [selectedFieldIdentifier, setSelectedFieldIdentifier] = useAtom(selectedFieldIdentifierAtom);
 
-    const selectedFieldIdentifier = useAtomValue(selectedFieldIdentifierAtom);
-    const setSelectedFieldIdentifier = useSetAtom(userSelectedFieldIdentifierAtom);
-    const [preferredViewLayout, setPreferredViewLayout] = useAtom(preferredViewLayoutAtom);
+    const [dataProviderSerializedState, setDataProviderSerializedState] = useAtom(dataProviderSerializedStateAtom);
+    const dataProviderSerializedStateRef = React.useRef(dataProviderSerializedState);
 
-    const persistState = React.useCallback(
+    const persistDataProviderManagerState = React.useCallback(
         function persistDataProviderManagerState() {
             if (!dataProviderManager) {
                 return;
             }
 
             const serializedState = {
                 dataProviderManager: dataProviderManager.serializeState(),
-                selectedFieldIdentifier,
-                preferredViewLayout,
             };
-            window.localStorage.setItem(
-                `${props.settingsContext.getInstanceIdString()}-settings`,
-                JSON.stringify(serializedState),
-            );
+            setDataProviderSerializedState(JSON.stringify(serializedState));
         },
-        [dataProviderManager, selectedFieldIdentifier, preferredViewLayout, props.settingsContext],
+        [dataProviderManager, setDataProviderSerializedState],
     );
 
     const applyPersistedState = React.useCallback(
```

Comment:
We have to discuss the update logic. When changing colorSet, this will reset the data provider manager and remove all added layers.